src/components/app/index.js

```diff
@@ -72,7 +72,10 @@ class App extends React.Component {
    * Dispatch an action to update the store with new pipeline data
    */
   updatePipelineData() {
-    const newState = preparePipelineState(this.props.data, true);
+    const newState = normalizeData(
```

Comment:
Seems a lot is happening in these few lines, could you explain in a comment a bit more about what is involved in this update?